### PR TITLE
REF: remove nav ctrl from providers

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

File: src/providers/push-notifications/push-notifications.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { FCM } from '@ionic-native/fcm';
-import { App, NavControllerBase } from 'ionic-angular';
```
Comment (msalcala11) [inline src/providers/push-notifications/push-notifications.ts]: Although not visible in this diff, it's a good idea to also remove the imported pages in this file.

File: src/pages/home/home.ts [hunk 2]
```diff
@@ -240,8 +259,43 @@ export class HomePage {
...
   private handleOpenUrl(url: string) {
-    if (!this.incomingDataProvider.redir(url)) {
+    this.incomingDataProvider.redir(url, 'HomePage').then((nextView) => {
+      this.redir(nextView);
+    }).catch(() => {
       this.logger.warn('Unknown URL! : ' + url);
+    })
+  }
+
+  private redir(nextView): void {
```
Comment (msalcala11) [inline src/pages/home/home.ts]: It looks like this code block is duplicated 3 times in this PR, which violates the DRY (Don't Repeat Yourself) principle. A more concise approach would be to define the switch statement on the AppComponent and to fire events from the incomingDataService to the AppComponent. The AppComponent would then perform the navCtrl.push action.

File: src/providers/push-notifications/push-notifications.ts (hunk 1 above)
Comment (gabrielbazan7) [reply src/providers/push-notifications/push-notifications.ts]: done! thanks

File: src/pages/home/home.ts (hunk 2 above)
Comment (gabrielbazan7) [reply src/pages/home/home.ts]: Agree. I just update de PR.

**Comment by msalcala11** [comment]
ACK!

**Comment by rastajpa** [comment]
ACK

**Comment by msalcala11** [comment]
Ooof codecov/patch under target. ACK rescinded XD

**Comment by cmgustavo** [comment]
Needs rebase

**Pull request merged**